ext/lib/mgmt/mcumgr/samples/smp_svr/zephyr/CMakeLists.txt

```diff
@@ -0,0 +1,45 @@
+# Top-level CMakeLists.txt for the skeleton application.
```

Comment:
We maintain samples under ZEPHYR_BASE/samples and we do build them on every commit, this is now being hosted under ext/ and is not being built and is not visible as a sample for someone to try, it also does not appear in the list of samples on the doc website.

If I try to build it btw, i get:

```
[ 94%] Built target kernel
/home/nashif/Work/github/zephyr/subsys/mgmt/buf.c: In function ‘cbor_nb_reader_init’:
/home/nashif/Work/github/zephyr/subsys/mgmt/buf.c:149:15: warning: assignment from incompatible pointer type [-Wincompatible-pointer-types]
  cnr->r.get32 = &cbor_nb_reader_get32;
               ^
/home/nashif/Work/github/zephyr/subsys/storage/flash_map/flash_map.c: In function ‘flash_area_get_sectors’:
/home/nashif/Work/github/zephyr/subsys/storage/flash_map/flash_map.c:191:32: warning: passing argument 2 of ‘flash_area_layout’ from incompatible pointer type [-Wincompatible-pointer-types]
  return flash_area_layout(idx, cnt, ret, get_sectors_cb, &data);
                                ^~~
/home/nashif/Work/github/zephyr/subsys/storage/flash_map/flash_map.c:136:12: note: expected ‘int *’ but argument is of type ‘uint32_t * {aka long unsigned int *}’
 static int flash_area_layout(int idx, int *cnt, void *ret,
            ^~~~~~~~~~~~~~~~~
```

We need to move this under samples/ and make it behave like other samples. @carlescufi  what do you think?

Reply:
Hmm... I was ignoring the flash_map warnings because I thought there was an outstanding PR to fix that.  It turns out I was thinking of a PR filed against MCUboot.  I can certainly fix the warnings, but I feel like that should go in a separate PR.

Re: the first warning (`cnr->r.get32`) - this is indeed a problem.  The checkpatch script spurred me to change occurrences of `uint32_t` to `u32_t`, but this breaks compatibility with the mcumgr API.  I think the correct fix is to change u32_t et. al. back to their standardized equivalents in these functions.

By the way, I am surprised you got this warning.  Your implementation doesn't use 32-bit `unsigned int`s?

Reply:
+1 to moving this sample into $ZEPHYR_BASE/samples

Reply:
The sample built out-of-the-box for me with nrf52_pca10040 :)

Reply:
I would also prefer to have a sample in `samples/` but the issue is that this sample comes from the mcumgr repository. One option would be **not** to import the mcumgr built-in sample into `ext/` and then have a sample in  `samples/` or, if we don't mind the redundancy, just creating a new sample in `samples/` and leave `ext/` as it is.

Reply:
I have not done it yet, but I will move the smp_svr sample out of `ext/lib/mgmt/mcumgr` and into the top-level `samples` directory.